embedded-graphics/src/primitives/circle.rs

```diff
@@ -274,11 +244,11 @@ where
     }
 }
 
-fn diameter_to_threshold(diameter: i32) -> i32 {
+fn diameter_to_threshold(diameter: u32) -> i32 {
```

Comment:
I think this should return a `u32`, i.e. `fn diameter_to_threshold(diameter: u32) -> u32` along with the other `i32` to `u32` fixes in this file to make it compile. The only change beyond type fixes would be that `DistanceIterator` becomes something like:

```diff
impl Iterator for DistanceIterator {
    type Item = (Point, u32);

    fn next(&mut self) -> Option<Self::Item> {
        self.points.next().map(|p| {
            let delta = self.center - p * 2;
-           let distance = delta.x.pow(2) + delta.y.pow(2);
+           let distance = delta.x.pow(2) as u32 + delta.y.pow(2) as u32;

            (p, distance)
        })
    }
}
```

(or maybe that should be `let distance = (delta.x.pow(2) as u32).saturating_add(delta.y.pow(2) as u32);` idk)

This makes it clearer to the reader that thresholds can never be negative, which percolates through the rest of the file.

Reply:
That's a good point. I've changed all thresholds and distances to `u32`.